lib/rdoc/rdoc.rb

```diff
@@ -232,6 +232,9 @@ def store= store
 
   def update_output_dir(op_dir, time, last = {})
     return if @options.dry_run or not @options.update_output_dir
+    unless ENV['SOURCE_DATE_EPOCH'].nil?
+      time = Time.at(ENV['SOURCE_DATE_EPOCH'].to_i).gmtime
```

Comment:
atm there is mostly the "generate" caller, if we do not count tests, so it could also be done there